lib/jose-jws-verify.js

```diff
@@ -159,16 +159,18 @@ JoseJWS.Verifier.prototype.addRecipient = function (rsa_key, key_id, alg) {
 /**
  * Verifies a JWS signature
  *
- * @returns Promise<Array> a Promise of an array  of objects { kid: string, verified: bool }
+ * @returns Promise<Array> a Promise of an array of objects { kid: string, verified: bool, payload?: string }
+ *
+ * payload is only populated and usable if verified is true
  */
 JoseJWS.Verifier.prototype.verify = function () {
 
   var that = this,
     signatures = that.signatures,
     key_promises = that.key_promises,
-      keyfinder = that.keyfinder,
+    keyfinder = that.keyfinder,
     promises = [],
-      check = !!keyfinder || Object.keys(that.key_promises).length > 0;
+    check = !!keyfinder || Object.keys(that.key_promises).length > 0;
```

Comment:
nit: these kinds of expressions are easier to read if written as `(!!keyfinder) || (Object.keys(that.key_promises).length > 0);